### PR TITLE
Added navigation buttons

### PR DESCRIPTION
**Pull request by Muntean-Ionut**

### CODE REVIEW

File: recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.css [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.dropdown{
+    position: relative;
+}
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.css]: I think this would've been better as in-line styling

File: recipe-box-frontend/src/main/frontend/src/components/Dropdown/DropdownButton/DropdownButton.css [hunk 2]
```diff
@@ -0,0 +1,23 @@
+.dropdown-btn {
+     display: flex;
+     align-items: center;
+     width: fit-content;
+     padding: 9px 28px;
+     background-color: #000000;
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Dropdown/DropdownButton/DropdownButton.css]: As a good rule-of-thumb, I recommend using multiples of 8 for padding. When it comes to buttons, I recommend having (top&bot padding)*2 = (left&right padding).

File: recipe-box-frontend/src/main/frontend/src/components/Dropdown/DropdownButton/DropdownButton.css [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+     border-radius: 8px;
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Dropdown/DropdownButton/DropdownButton.css]: Why is this needed? The button's border is practically invisible.

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx [hunk 4]
```diff
@@ -23,23 +23,16 @@ const Header: React.FC = () => {
...
+                                    <li className='left'>
+                                    <Dropdown buttonText="Recipes" content={
+                                        <>
+                                            <Link to="/recipes/list" className="dropdown-item">Public recipes</Link>
+                                            <Link to="/recipes/myRecipes" className="dropdown-item">My recipes</Link>
+                                            <Link to="/recipes/ingredients" className="dropdown-item">Recipes with my ingredients</Link>
+                                        </>
+                                    }/>
+                                    </li>
+                                </>
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: Here you are violating the DOM nesting rules. An "li" should always be nested inside an "ul" or an "ol", never inside another "li". I think you should refactor this.

Also, we are using "useNavigate" throughout our app. For consistency, you should use it here too.

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx [hunk 5]
```diff
@@ -55,9 +48,21 @@ const Header: React.FC = () => {
...
                     <div className="right-items">
+                        {userType === UserType.GUEST &&
                         <li className="right">
                             <Link to="/userlogin" className="button">LOGIN</Link>
                         </li>
+                        }
+                        {userType !== UserType.GUEST &&
+                        <>
+                        <li className="right">
+                            <Link to="" className="button">ADD NEW RECIPE</Link>
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: This here doesn't take us where it should.

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.css [hunk 6]
```diff
@@ -38,6 +38,10 @@
     height: 25px;
 }
 
+.profile-logo-image {
+    height: 25px;
+}
+
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.css]: Pro tip: You should use Optical Compensation here: make the circle's diameter 5% to 10% bigger than the addRecipe's button's height.

Also, the two elements are too close to each other. Since they are not related in function, they should feel a bit more separated.

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.css [hunk 7]
```diff
@@ -51,6 +55,8 @@
     border-radius: 6px;
     border: 2px solid #fff;
     font-size: 12px;
+    font-weight: bold;
+
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.css]: When the viewport gets too narrow, the button gets sliced up, i suggest using `white-space: nowrap;` for it and for Meal Plans.

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.css [hunk 8]
```diff
@@ -109,7 +100,7 @@
 }
 
 .dropdown-item:hover {
-    background-color: #f1f1f1;
+    font-weight: bold;
```
Comment (AlinFlorescu80) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.css]: Here the design showed a changing background color and not a font-weight.

File: recipe-box-backend/src/main/resources/application.properties
Comment (Ionel14) [inline recipe-box-backend/src/main/resources/application.properties:1]: I don't think this file should be added

File: recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        }
+    }
+    window.addEventListener("click",handleClickOutsideDropdown);
+
+    return (
+           <div style={{position: 'relative'}} ref={dropdownRef}>
+              <DropdownButton onClick={() => setOpen(!open)} open={open}>
+                  {buttonText}
+              </DropdownButton>
+              <DropdownContent open={open}>
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.tsx:33]: I suggest to render the content conditionally, only if it's open

File: recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+    const [open, setOpen] = useState(false);
+
+    const dropdownRef = useRef<HTMLDivElement>(null);
+
+    const handleClickOutsideDropdown =(e:any)=> {
+        if(open && !dropdownRef.current?.contains(e.target as Node)){
+            setOpen(false);
+        }
+    }
+    window.addEventListener("click",handleClickOutsideDropdown);
```
Comment (en-alexandra-costache) [inline recipe-box-frontend/src/main/frontend/src/components/Dropdown/Dropdown.tsx]: Nice touch, but it would be better to wrap this into an useEffect along with the handleClickOutsideOpen function in order to avoid the browser adding the event listener at every render. Set the dependencies for the useEffect to be `dropdownRef` and `open`.
Also, in the useEffect function, return a function that removes the event listener you're setting.
It should look something like this:
image

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx [hunk 11]
```diff
@@ -23,41 +24,46 @@ const Header: React.FC = () => {
...
-                                            <li><Link to="/recipes/list" className="dropdown-item">Public Recipes</Link></li>
-                                            <li><Link to="/recipes/myRecipes" className="dropdown-item">My Recipes</Link></li>
-                                            <li><Link to="/recipes/ingredients" className="dropdown-item">Recipes with My Ingredients</Link></li>
-                                        </ul>
-                                        )}
-                                    </>
+                                    <ul className='left'>
+                                    <Dropdown buttonText="Recipes" content={
+                                        <>
+                                            <button className="dropdown-item" onClick={() => navigate("/recipes/list")}>Public recipes</button>
```
Comment (en-alexandra-costache) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: Why use navigate when you could use the Link component?

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx [hunk 12]
```diff
@@ -23,41 +24,46 @@ const Header: React.FC = () => {
...
                                 
                                 {userType === UserType.CHEF && (
-                                    <li><Link to="/meal-plans" className="nav-item">Meal Plans</Link></li>
+                                    <li><button className="nav-item" onClick={() => navigate("/meal-plans")}>Meal Plans</button></li>
                                 )}
                             </>
                         )}
                     </div>
                     <div className="right-items">
+                        {userType === UserType.GUEST &&
```
Comment (en-catrinel-lupu) [inline recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: I see we have one condition if the user type is chef and one is the user type is different from chef. You can improve this by using the ternary operator

**Review by en-alexandra-costache — CHANGES_REQUESTED** [review]

File: recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx (hunk 11 above)
Comment (Muntean-Ionut) [reply recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: Because Alin said in a review: "Also, we are using "useNavigate" throughout our app. For consistency, you should use it here too.", so i changed the links to be "useNavigate" instead. Should i change them back?

Comment (AlinFlorescu80) [reply recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: Did I do an oopsie? :)))

Comment (en-alexandra-costache) [reply recipe-box-frontend/src/main/frontend/src/components/Header/Header.tsx]: :)) There aren't such things as oopsies in programming. While you are right that it would be recommended to keep in consistency, the Link element would be a more preferable method than useNavigate for navigating to the pages. You can eventually use useNavigate on the pages that should be accessed only by a specific group of users in order to redirect those who shouldn't have access to the home page.
@en-marius-gociman , @en-catrinel-lupu, correct me if I'm wrong

**Review by en-catrinel-lupu — APPROVED** [review]

**Review by en-alexandra-costache — APPROVED** [review]
